### ISSUE TITLE
Request clarification on the meaning of "sample"

### ISSUE THREAD

**Issue by VisLab** [bids-standard/bids-specification#1043]
The [Task Events](https://github.com/bids-standard/bids-specification/blob/master/src/04-modality-specific-files/05-task-events.md) section lists "sample" as an optional integer entry, with the description:

> Onset of the event according to the sampling scheme of the recorded modality (that is, referring to the raw data file that the events.tsv file accompanies).

Could someone please clarify this?  Does it mean sample number?  Does it start at 0 or at 1?

**Comment by sappelhoff** [comment]
>  Does it mean sample number? 

Yes, for example if you want the exact data of an EEG recording with a sampling frequency of 1000Hz at 1 second, you would specify `sample = 1000`

>  Does it start at 0 or at 1?

this is related to https://github.com/bids-standard/bids-specification/issues/499 -- unfortunately there is no explicit indication of zero versus one-based indexing in BIDS at the moment.

In my short example above I used one-based indexing ... but in the #499 thread is a good argument for zero-based indexing in my opinion.

**Comment by Remi-Gau** [comment]
@sappelhoff 
is this documented anywhere because it is true that the column description is not super clear?
had to reverse engineer what it could mean from one the bids example.

Should we clarify this definition with an example?

**Comment by VisLab** [comment]
This clarification might not be as easy as it sounds. For example, European Data Format (EDF) which seems to be a preferred format for BIDS EEG is a multi-channel file format that allows different sampling rates for different channels in the same file.

**Comment by sappelhoff** [comment]
Right. If I could turn back time, I'd maybe not make `sample` a "first class citizen" in the BIDS spec, and instead would have used it as one (of many) examples of the arbitrary columns that users can define themselves.

But now that we are stuck with it, I am up for improving the text, including the caveat raised by Kay. One could for example discourage the use of `sample` if there are more than one relevant sampling frequencies.

**Comment by VisLab** [comment]
Closely related to the sample issue is how many significant figures should be provided for `onset` and `duration` in the `events.tsv` files.

The event files on openNeuro range from integers to numbers with 16 significant figures.
There seems to be little relationship to the actual accuracy of the values.
Anyone writing tools to clean up event files as part of the dataset formatting for BIDS
will likely end up rewriting/regenerating the event files and the decision has to be made.

Does anyone have any guidance on this issue?

**Comment by sappelhoff** [comment]
> Closely related to the sample issue is how many significant figures should be provided for onset and duration in the events.tsv files.

As you have probably seen, we don't have any requirements for this ... I would be open to adding a reasonable recommendation, if you have one in mind.

>  There seems to be little relationship to the actual accuracy of the values.

I don't understand what you mean by that.

> Does anyone have any guidance on this issue?

as mentioned above: I think we could add a recommendation to the spec text, but I wouldn't know how that could look like, apart from maybe:

> we RECOMMEND that dataset curators specify "onset" and "duration" with as much decimal precision as is reasonable in the context of the experiment. For example in an EEG experiment with devices operating at 1000Hz sampling frequency, dataset curators SHOULD specify at least 3 figures after the decimal point.
